modules/gateway/peersmanager.go

```diff
@@ -31,6 +31,11 @@ func (g *Gateway) managedPeerManagerConnect(addr modules.NetAddress) {
 			// race condition could mean that the peer was disconnected
 			// before this code block was reached.
 			p.Inbound = false
+			if n, ok := g.nodes[p.NetAddress]; ok && !n.WasOutboundPeer {
+				n.WasOutboundPeer = true
+				g.nodes[n.NetAddress] = n
+				g.saveSync()
```

Comment:
don't need to saveSync for automated actions, it'll get picked up by the save loop after `saveFrequency` anyway.